src/generator/docker-compose-file-generator.ts

```diff
@@ -1,40 +1,165 @@
+     * @param dockerCompose - Docker Compose configuration object.
+     * @param runnableServices - List of services to include.
+     * @param hasExcludedServices - Indicates if there are excluded services.
+     * @returns Updated Docker Compose configuration.
+     */
+    private addIncludePropertiesToMainDockerCompose (dockerCompose:DockerComposeSpec, runnableServices:ServiceWithLiveUpdate[], hasExcludedServices = false): DockerComposeSpec {
+        if (!hasExcludedServices) {
+            dockerCompose.include = runnableServices.map(
+                service => service.liveUpdate
+                    ? join(this.path, "local", service.name, "docker-compose.yaml")
+                    : service.source
+            ).reduce(deduplicate, []);
+        } else {
+            const includes: string[] = [];
+            runnableServices.forEach(service => {
+                if (service.liveUpdate) {
+                    includes.push(join(this.path, "local", service.name, "docker-compose.yaml"));
+                }
+            });
+            const otherDocker = getAllFilesInDirectory(join(this.path, "exclusion-filtered-compose-files"));
+            includes.push(...otherDocker);
+            dockerCompose.include = includes;
+        }
+        return dockerCompose;
+    }
+
+    /**
+     * Removes excluded services from the `depends_on` property of each service in the Docker Compose configuration.
+     * Handles different formats of `depends_on`: array format and object format.
+     * @param dockerComposeConfig - Docker Compose configuration object.
+     * @param excludedServices - List of excluded service names.
+     * @returns Updated Docker Compose configuration.
+     */
+    private removeExcludedServiceFromDependOn (dockerComposeConfig: DockerComposeSpec, excludedServices: string[]): DockerComposeSpec {
+        for (const serviceSpec of Object.values(dockerComposeConfig.services)) {
+            if (serviceSpec.depends_on) {
+                // Handle object format
+                if (typeof serviceSpec.depends_on === "object" && !Array.isArray(serviceSpec.depends_on)) {
+                    excludedServices?.forEach((excludedService) => {
+                        if (excludedService in serviceSpec.depends_on!) {
+                            delete serviceSpec.depends_on![excludedService];
+                        }
+                    });
+                }
+
+                // Handle array format
+                if (Array.isArray(serviceSpec.depends_on)) {
+                    const includes = serviceSpec.depends_on.filter(
+                        (dependency) => !excludedServices?.includes(dependency)
+                    );
+                    if (includes.length > 0) {
+                        serviceSpec.depends_on = includes;
+                    } else {
+                        delete serviceSpec.depends_on;
+                    }
+                }
+            }
+        }
```

Comment:
Quite complicated block of code, effectively have a loop within a loop, consider whether this can be simplified by extracting methods or using `map`/`filter` sort of thing to compose it as a pipeline.